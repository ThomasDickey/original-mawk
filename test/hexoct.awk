#
# test that hex and octal numbers are converted correctly
#
BEGIN {
    xa = 055;
    xb = 0x45;
    printf "%d %x %o %c\n", 0x45x2, 0x45x2, 0x45x2, 0x45x2;
    printf "%d %x %o %c\n", xa, xa, xa, xa;
    printf "%d %x %o %c\n", xb, xb, xb, xb;
    printf "%d %x %o %c\n", 055v2, 055g2, 055j2, 055x2;

    printf "\n";
    printf "%d %x %o %c\n", +xa, +xa, +xa, +xa;
    printf "%d %x %o %c\n", -xa, -xa, -xa, -xa;

    printf "\n";
    xa = -055;
    xb = -0x45;
    printf "%d %x %o %c\n", xa, xa, xa, xa;
    printf "%d %x %o %c\n", xa, xa, xa, xa;
    printf "%d %x %o %c\n", -055v2, -055g2, -055j2, -055x2;
}
