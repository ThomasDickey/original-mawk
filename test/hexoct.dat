0x45
0x45vv
-055
