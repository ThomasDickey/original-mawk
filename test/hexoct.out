69 45 105 E
45 2d 55 -
69 45 105 E
45 2d 55 -

45 2d 55 -
-45 ffffffd3 37777777723 Ó

-45 ffffffd3 37777777723 Ó
-45 ffffffd3 37777777723 Ó
45 2d 55 -
69 45 105
69 45 105
45 2d 55
